src/main/scala/org/scalameter/PerformanceTest.scala

```diff
@@ -38,11 +37,15 @@ object PerformanceTest {
 
     type SameType
 
-    def executeTests(): Boolean = {
+    final def executeTests(): Boolean = {
       val datestart: Option[Date] = Some(new Date)
-      val rawsetuptree = BasePerformanceTest.setupzipper.value.result
+      val initialContext = executor.measurer.prepareContext(currentContext ++ defaultConfig)
+      val rawsetuptree = BasePerformanceTest.setupzipper.value.setContext(initialContext).result.map { setup =>
```

Comment:
This is not doing the right thing - the `setupzipper` initialization process is a bit tricky, and is relying heavily on dynamic variables to achieve the nice dsl.

Take a look at the `dsl.scala` file. The parts of the tree in the `setupzipper` are initialized exactly when the various `performance of` and `using` blocks are called. This happens during the construction of the `BasePerformanceTest` subclass object, which is _before_ `executeTests` gets called.
When `executeTests` gets called, the setup zipper is already constructed, and has all the contexts and `Setup` values populated.

You must ensure that the initial value of the `setupzipper` is set correctly in the `BasePerformanceTest` constructor before the (user-defined) subclass constructors get called. By correct, I mean - the setup zipper must contain only empty the root tree value (`Tree.Zipper.root`) and the context of this root must be set to:

    prepareContext(currentContext ++ defaultConfig)

My advice is also that you change the `Tree.Zipper.root` factory method to take the `initialContext` argument, instead of relying on a dynamic variable.